node/src/components/linear_chain_sync/operations.rs

```diff
@@ -113,6 +113,66 @@ where
     Ok(block_header)
 }
 
+/// Get trusted block headers; falls back to genesis if none are available
+async fn get_trusted_validator_weights<REv, I>(
+    effect_builder: EffectBuilder<REv>,
+    chainspec: &Chainspec,
+    trusted_header: &BlockHeader,
+) -> Result<BTreeMap<PublicKey, U512>, LinearChainSyncError<I>>
+where
+    REv: From<FetcherRequest<I, BlockHeader>> + From<NetworkInfoRequest<I>> + From<StorageRequest>,
+    I: Eq + Debug + Clone + Send + 'static,
+{
+    // If we are right after genesis, use the genesis validators.
+    if trusted_header.era_id() == EraId::new(0) {
+        let genesis_validator_weights: BTreeMap<PublicKey, U512> = chainspec
+            .network_config
+            .accounts_config
+            .accounts()
+            .iter()
+            .filter_map(|account_config| {
+                if account_config.is_genesis_validator() {
+                    Some((
+                        account_config.public_key(),
+                        account_config.bonded_amount().value(),
```

Comment:
Does that include delegators? (Not sure how that works at genesis.)

Reply:
I _think_ so?  Paging @Fraser999 ... maybe we should have a function that shares this logic in both the genesis code and here as well.